asn1ate/sema.py

```diff
@@ -302,6 +307,12 @@ def resolve_type_decl(self, type_decl, referenced_modules):
                 raise Exception('Unrecognized referenced module %s in %s.' % (type_decl.module_ref.name,
                                                                               [module.name for module in
                                                                                referenced_modules]))
+
+            if type_decl.type_name not in module.user_types():
+                if skip_missing:
+                    for k, value in self.imports.imports.items():
```

Comment:
that's the idea, we skip IFF the declared type was in the IMPORTS section of the ASN.1 schema otherwise it will keep the same behaviour as before, e.g. raising an KeyError exception.

Reply:
I don't really understand _why_, though. It's been a couple of years since I had my head in ASN.1, so I'm a little rusty...

I can see that non-imported types (defined in the same module) should never be missing, but I'm not sure why it's more OK if it's imported.

Reply:
maybe this would solve a use case like: no correct multi-module schema is available and the user still wants to generate a code that he can after manually change to resolve the imports?

Reply:
Yeah, I think that makes sense. I wonder if you could substitute something like ASN.1 `ANY` to indicate that there's a hole in the schema, and print a warning that the original type is unknown.

Reply:
what I can think of is to collect together all the missing referenced identifier (from all the modules in the _corpus_) and add them to the end of the generated output; e.g. if  IPAddress is referenced in the schema but imported from a missing ASN.1 (let's call it **missingschema**) , then `asn1ate` could generate:


```
try:
  from missingschema import IPAddress
except ImportError:
  from pyasn1.type import univ
  def  IPAddress(univ.Any):
    pass
```

Reply:
Shouldn't you be able to know that `missingschema` is unavailable already at codegen time? Doesn't seem very useful to force an `ImportError`. But on the other hand it's actually a nice visual indicator in the generated code that something is off.

Reply:
sorry for the delay, I had some urgencies..
did you make your mind clear about this?

Reply:
No, I've also been busy with other things. It would be nice to just conclude on something, but I think it takes some iteration to figure out what the best solution is.